### ISSUE TITLE
MSYS2 version does not open files with tildes in filenames (Unicode)

### ISSUE THREAD

**Issue by dustwanderer** [ImageMagick/ImageMagick#5680]
### ImageMagick version

7.1.0-50 beta Q16-HDRI x86_64

### Operating system

Windows

### Operating system, version and so on

Windows 11 Version 21H2 (OS build 22000.1098)

### Description

Hi,
I had been using ImageMagick binaries for Windows. Recently, I switched to the [MSYS2 version of ImageMagick](https://packages.msys2.org/package/mingw-w64-ucrt-x86_64-imagemagick?repo=ucrt64), because the MSYS2 version includes the FFT delegate. However, the MSYS2 version of ImageMagick can't open files whose filename or filepath include tildes. The Windows version of ImageMagick  7.1.0-51 Q16-HDRI x64 can open those files.

Both Windows and MSYS2 are up-to-date.

Thanks for a great program and best regards

### Steps to Reproduce

I open a CMD command line window. I type `ChCP 65001` to set Unicode UTF-8 as the CMD shell language. I also set the environment variables "LANG"," LANGUAGE", and "LC_ALL" as "en_US.UTF-8". The command `magick.exe identify páramo.jpg` fails with the error message "identify: unable to open image 'páramo.jpg': No such file or directory @ error/blob.c/OpenBlob/3570.".

Other MSYS2 commands work fine. For example `file.exe páramo.jpg` can open the target file.

The command `magick.exe identify 01.jpg` works fine. However, the command `magick.exe identify C:\árbol\01.jpg` fails, although it is the same file. `magick.exe identify "C:\árbol\01.jpg"` also fails.

If I use the MSYS2 bash shell (or Windows PowerShell) MSYS2 ImageMagick also fails.

**Comment by fmw42** [comment]
The tilde (~) is a Linux "shortcut" to denote a user's home directory (when at beginning of name). See https://twiki.org/cgi-bin/view/Wikilearn/TildeSlash

**Comment by dustwanderer** [comment]
Sorry. I wrote "tildes" when I should written "diacritics" or "accent marks". In Spanish we call them tildes and I absentmindedly forgot that the English word is different.

The issue that I should have reported is that the MSYS2 version of ImageMagick can't open files whose filenames contain diacritics. I assume that the same problem happens with files whose filenames contain non-standard ASCII characters

**Comment by dlemstra** [comment]
The ImageMagick team did not create the MSYS2 version of ImageMagick and this might be something in their patches or their build process. I would advise you to reach out to them and ask them why this is not working and if this is something inside our project then they can reach out to us. But maybe @snibgo knows why this is happening?

And we do not include the FFT delegate by default due to licensing issues. It is licensed under GPL and requires a commercial license if you want to use it in a commercial project. The MSYS project page should not provide this due to licensing issues? FFTW is GPL 2.0 and that is not compatible with our Apache 2 license?

You can also create your own build of ImageMagick if you follow these instructions: https://github.com/ImageMagick/ImageMagick-Windows. In the `configure.exe` tool there is an option to build projects that are not compatible with the Apache 2.0 license.

**Comment by snibgo** [comment]
With IM version 7.1.0-20 built with Cygwin tools, and 7.1.0-49 pre-built binary, on Windows 8.1, command window, I don't have this problem. For example:
```
f:\web\im>%IMG7%magick rose: páramo.jpg

f:\web\im>%IMG7%magick identify páramo.jpg
páramo.jpg JPEG 70x46 70x46+0+0 8-bit sRGB 2664B 0.000u 0:00.001
```
I don't use MSYS2 and can't comment on that.

**Comment by dustwanderer** [comment]
Thank you for your help. I have submitted the issue to the forum for MSYS2 packages.

Best regars

**Comment by Biswa96** [comment]
This issue has been fixed in [7.1.0-52](https://github.com/ImageMagick/ImageMagick/releases/tag/7.1.0-52) version with a91d45bbe3a902d816fdebb62a8eb5d6f79132f3 commit.
